packages/pigeon/lib/cpp_generator.dart

```diff
@@ -679,8 +736,19 @@ String _getSafeArgumentName(int count, NamedType argument) =>
 
 void _writeFlutterApiHeader(Indent indent, Api api) {
   assert(api.location == ApiLocation.flutter);
-  indent.writeln(
-      '/* Generated class from Pigeon that represents Flutter messages that can be called from C++. */');
+
+  const String generatedMessage =
+      'Generated class from Pigeon that represents Flutter messages that can be called from C++.';
+  if (api.documentationComments != null) {
+    indent.writeln(openDoc);
+    indent.writeln(generatedMessage);
+    api.documentationComments?.forEach((String documentationComment) {
+      indent.writeln(documentationComment);
+    });
+    indent.writeln(closeDoc);
+  } else {
+    indent.writeln('$openDoc $generatedMessage $closeDoc');
+  }
```

Comment:
This structure is duplicated across the code.  I'd factor out a helper function, `_writeDocstring`?